pulsar-client-cpp/include/pulsar/ProducerConfiguration.h

```diff
@@ -69,6 +69,9 @@ class ProducerConfiguration {
     ProducerConfiguration& setMessageRouter(const MessageRoutingPolicyPtr& router);
     const MessageRoutingPolicyPtr& getMessageRouterPtr() const;
 
+    ProducerConfiguration& setUseMurmurHash(const bool useMurmurHash);
```

Comment:
I would prefer to have an enum type to select the default hashing scheme (default, since it can always be overridden in a custom message router.

Something like: 

```cpp
ProducerConfiguration& setDefaultHashingScheme(HashingScheme hashingScheme);
```

This way it might be interesting to provide the same function across C++ and Java, even the existing functions (eg: `JavaStringHash`)